BaseAgent.py

```diff
@@ -191,14 +204,23 @@ def learn(self, total_timesteps: int):
                         train_time = (time.thread_time_ns() - init_train_time) / 1e9
                         train_fps = self.log_interval / train_time
                         self.log_history('time/train_fps', train_fps, self.learn_env_steps)
-                        self.avg_eval_rwd = self.evaluate()
+                        if self.use_threaded_eval:
```

Comment:
looks like we are waiting for the finish of the old worker but not starting the new worker thread when `self.use_threaded_eval` is true